lib/Alchemy/Phrasea/Twig/PhraseanetExtension.php

```diff
@@ -37,9 +37,44 @@ public function getFunctions()
             new \Twig_SimpleFunction('collection_logo', array($this, 'getCollectionLogo'), array(
                 'is_safe' => array('html')
             )),
+            new \Twig_SimpleFunction('record_flags', array($this, 'getRecordFlags'))
         );
     }
 
+    public function getRecordFlags(RecordInterface $record)
+    {
+        $databox =  $this->app['phraseanet.appbox']->get_databox($record->getDataboxId());
+
+        $structure = $this->app['factory.status-structure']->getStructure($databox);
+
+        $user  = $this->app['authentication']->getUser();
+
+        if (false === ($user instanceof User)) {
+            return $structure->getStatuses();
+        }
+
+        $acl = $this->app['acl']->get($user);
+
+        $status = $structure->getStatuses();
+
+        if (false === $acl->has_right_on_base($record->getBaseId(), 'chgstatus')) {
+            $status = $status->displayable();
+        }
+
+        $bitMask = $record->getStatusBitMask();
+
+        $status = $status->illustrated($bitMask);
```

Comment:
IMO, `illustrated()` and `displayable()` and all computing of this method should be encapsulated in the status structure or the record (eventually delegating to the status structure).